src/sys.zig

```diff
@@ -349,42 +349,48 @@ pub fn getErrno(rc: anytype) bun.C.E {
     };
 }
 
-// pub fn openOptionsFromFlagsWindows(flags: u32) windows.OpenFileOptions {
-//     const w = windows;
-//     const O = std.os.O;
-
-//     var access_mask: w.ULONG = w.READ_CONTROL | w.FILE_WRITE_ATTRIBUTES | w.SYNCHRONIZE;
-//     if (flags & O.RDWR != 0) {
-//         access_mask |= w.GENERIC_READ | w.GENERIC_WRITE;
-//     } else if (flags & O.WRONLY != 0) {
-//         access_mask |= w.GENERIC_WRITE;
-//     } else {
-//         access_mask |= w.GENERIC_READ | w.GENERIC_WRITE;
-//     }
-
-//     const filter: windows.OpenFileOptions.Filter = if (flags & O.DIRECTORY != 0) .dir_only else .file_only;
-//     const follow_symlinks: bool = flags & O.NOFOLLOW == 0;
-
-//     const creation: w.ULONG = blk: {
-//         if (flags & O.CREAT != 0) {
-//             if (flags & O.EXCL != 0) {
-//                 break :blk w.FILE_CREATE;
-//             }
-//         }
-//         break :blk w.FILE_OPEN;
-//     };
-
-//     return .{
-//         .access_mask = access_mask,
-//         .io_mode = .blocking,
-//         .creation = creation,
-//         .filter = filter,
-//         .follow_symlinks = follow_symlinks,
-//     };
-// }
 const O = std.os.O;
 const w = std.os.windows;
 
+fn normalizePathWindows(
+    dir_fd: bun.FileDescriptor,
+    path: []const u8,
+    buf: *bun.WPathBuffer,
+) Maybe([:0]const u16) {
+    // TODO this check can be more sophisticated: paths like C:\bun\hello.txt are fine, even though they contain a dot
+    if (!bun.strings.containsChar(path, '.')) {
```

Comment:
I think this needs to be more sophisticated before we can realistically ship this PR. This PR will add too many  system calls.

```suggestion
    if (bun.strings.containsRelativePath(path)) {
```

Something like this:
```zig
pub fn  containsRelativePath(input: []const u8) bool {
   const dot = strings.indexOfCharUsize(input, '.') orelse return false;
   // "."
   // "/."
   // ".."
   if (dot == 0 or dot == input.len - 1)
     return true;
 // - "/foo\bar/../"
 // - "/foo\bar/."
 // - "/../"
 // - "/foo/bar/baz/../"
  if (isAnySep(input[dot - 1]) 
     return true;

     return false;
}
```